daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -381,12 +390,16 @@ def run(self):
 
         # Inputs are un-pickled and treated as the arguments of the function
         # Their order must be preserved, so we use an OrderedDict
-        if self.pickle:
+        if DropParser(self.input_parser) is DropParser.PICKLE:
```

Comment:
I cast again to DropParser throughout the class as the initializing of app args to class members is done after the dlg_x_param definitions are evaluated in the parent constructor. This order is problematic as it means apps have no control over the member types as the param definitions are overwritten at this point.

In this case `input_parser`string is read from kwargs and using dlg_x_param reassigns to DropParser, application args contains a string literal and reassigns it back to string.

The app param logic should be processed with component params in AbstractDrop._extract_attributes, perhaps to self.appArgs